client/src/components/campaignCreatorComponents/CampaignCreator.tsx

```diff
@@ -36,8 +43,81 @@ const CampaignCreator = ({ campaignSettings, setCampaignSettings }: CampaignCrea
         navigate('/twilight-library/dashboard');
     };
 
+    const addToTimeline = (
+        timeline: Record<number, string[]>,
+        nodeKey: keyof NodePillarLists,
+        typeKey: keyof TimelineOptionList,
+    ) => {
+        if (!selectedCampaign.flexible_nemesis_encounters && typeKey === 'nemesis') {
+            // Handle strict Nemesis timeline
+            const nemesis_tier: string = nodeKey.slice(-1);
+            selectedCampaign[nodeKey].forEach((selection: string) => {
+                [1, 2, 3].forEach((level: number) => {
+                    const fight_str: string = `nn${nemesis_tier}_lvl${level}_fight_year`;
+                    const fight_key: keyof NemesisFightYearLists = fight_str as keyof NemesisFightYearLists;
+                    const fight_year: number | null = selectedCampaign[fight_key];
```

Comment:
Very nitpicky, but the type definitions here subtract more from the legibility of these assignments than they add to your devlife imo.

The scope of these vars is so short, and TypeScript will handle a lot of type inference to yell at you when you need it such that my personal opinion is that the specificity muddies the waters when trying to read.
Mileage may vary

Reply:
Rarely do you ever want to declare your types when you set a type, instead let typescript infer

Reply:
ok noted